### PR TITLE
Simple dump speedup

### PR DESCRIPTION
**Pull request by graetzer**
Same as https://github.com/arangodb/arangodb/pull/5274

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBReplicationContext.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ using namespace arangodb::velocypack;
 namespace {
 TRI_voc_cid_t normalizeIdentifier(transaction::Methods const& trx,
                                   std::string const& identifier) {
-  TRI_voc_cid_t id{0};
+  TRI_voc_cid_t id = 0;
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBReplicationContext.cpp]: Why?

File: arangod/RocksDBEngine/RocksDBReplicationContext.cpp [hunk 2]
```diff
@@ -270,62 +268,106 @@ RocksDBReplicationResult RocksDBReplicationContext::dump(
     if (!_trx || !_guard || (&(_guard->database()) != vocbase)) {
       return RocksDBReplicationResult(TRI_ERROR_BAD_PARAMETER, _lastTick);
     }
-
-    TRI_voc_cid_t const id{::normalizeIdentifier(*_trx, collectionName)};
-
-    if (0 == id) {
+    TRI_voc_cid_t const cid = ::normalizeIdentifier(*_trx, cname);
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBReplicationContext.cpp]: Why?

File: arangod/RocksDBEngine/RocksDBReplicationContext.h [hunk 3]
```diff
@@ -46,19 +47,29 @@ class RocksDBReplicationContext {
...
     arangodb::velocypack::Options vpackOptions;
+    
+    bool sorted() const { return sortedIterator; }
+    void setSorted(bool, transaction::Methods*);
 
     void release();
+    
+  private:
+    /// primary-index sorted iterator
+    bool sortedIterator;
```
Comment (danielhlarkin) [inline arangod/RocksDBEngine/RocksDBReplicationContext.h]: `_sortedIterator` for convention?

File: arangod/VocBase/Methods/Indexes.cpp [hunk 4]
```diff
@@ -128,8 +128,11 @@ arangodb::Result Indexes::getAll(LogicalCollection const* collection,
...
       } else {
         tmp.openObject();
-        for(auto const& i : VPackObjectIterator(s)){
-          tmp.add(i.key.copyString(), i.value);
+        for (auto const& it : VPackObjectIterator(s, true)) {
+          VPackValueLength l;
+          char const* p = it.key.getString(l);
+          tmp.add(VPackValuePair(p, l, VPackValueType::String));
+          tmp.add(it.value);
         }
```
Comment (danielhlarkin) [inline arangod/VocBase/Methods/Indexes.cpp]: Do these changes save a key copy?

File: arangod/RocksDBEngine/RocksDBReplicationContext.cpp (hunk 1 above)
Comment (graetzer) [reply arangod/RocksDBEngine/RocksDBReplicationContext.cpp]: because its just a uint64_t ?

**Pull request merged**